build.sbt

```diff
@@ -16,7 +16,9 @@ resolvers ++= Seq(
 
 libraryDependencies ++= Seq(
   "org.scalatest" %% "scalatest" % "3.2.2",
-  "com.lihaoyi" %% "utest" % "latest.integration"
+  "com.lihaoyi" %% "utest" % "latest.integration",
+  "com.github.plokhotnyuk.jsoniter-scala" %% "jsoniter-scala-core"   % "2.6.4",
+  "com.github.plokhotnyuk.jsoniter-scala" %% "jsoniter-scala-macros" % "2.6.4" % "provided"
```

Comment:
I don't think we should be adding a second json library to our dependencies.
Could you check to see how hard it would be to use [json4s](https://github.com/json4s/json4s) instead?
That is the library used by `firrtl` and thus already pulled into `chiseltest`.

Reply:
For now I've kept jsoniter because I've found its feature set is better set-up for serializing generic objects without having to write extra converters. (I.E as it is I can serialize the CombinationPath object without the need to write anything special). I haven't been able to replicate that with json4s but I'll keep investigating.

Reply:
I don't want to sign off on adding a new dependency. Removing that dependency later will be a great pain, since users might start depending on it. If @ducky64  is OK, with adding it, then we can. However, if it up to just me, then I believe that we should not add this dependency.

Reply:
I generally would prefer to not add dependencies, and additionally prefer to standardize libraries across the chisel projects where possible. But if there's some strong rationale for a new library (or if we want to standardize on a new library project-wide, perhaps without necessarily migrating old code), it's worth discussing.

Reply:
We're fairly close to being able to build Chisel and FIRRTL without any runtime reflection except for generic serialization and deserialization of annotations. This isn't like, an important goal, but it would be cool to be able to emit ScalaJS as well as create GraalVM Native images. jsoniter appears to have better support for non-JVM targets than json4s, so it's worth considering using it throughout instead. This is a high code migration cost though since some users have written custom [de]serializers.